### PR TITLE
Add `deviceRegistry.type=memory` deprecation message

### PR DESCRIPTION
**Pull request by mapedraza**
Should other parts of the doc edited? 

I.E:

https://github.com/telefonicaid/iotagent-ul/blob/b7e3099e3eb08dc03f45f1f2d5595f6381326971/docker/README.md?plain=1#L19-L27

### CODE REVIEW

File: doc/deprecated.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ A list of deprecated features and the version in which they were deprecated foll
 -   Support to Node.js v10 in iotagent-node-lib 2.15.0 (finally removed in 2.16.0)
 -   Support to Node.js v12 in iotagent-node-lib 2.24.0 (finally removed in 2.25.0)
 -   Support to NGSI-LD v1.3 in iotagent-node-lib 2.25.0
+-   Support to `deviceRegistry.type=memory` value.
```
Comment (fgalan) [inline doc/deprecated.md:22]: Add also another item for the groups statically defined by configuration

Comment (mapedraza) [reply doc/deprecated.md]: Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1318/commits/8a79b4a12949dd378b26d71124961f87d6a0d0a9

File: doc/deprecated.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ A list of deprecated features and the version in which they were deprecated foll
 -   Support to Node.js v10 in iotagent-node-lib 2.15.0 (finally removed in 2.16.0)
 -   Support to Node.js v12 in iotagent-node-lib 2.24.0 (finally removed in 2.25.0)
 -   Support to NGSI-LD v1.3 in iotagent-node-lib 2.25.0 (finally removed in 2.26.0)
+-   Support groups (provision) statically defined by configuration.
+-   Support to `deviceRegistry.type=memory` value.
```
Comment (fgalan) [inline doc/deprecated.md:23]:
```suggestion
-   Support groups (provision) statically defined by configuration
-   Support to in-memory registry (i.e.`deviceRegistry.type=memory`)
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**